### DISCUSSION TITLE
Can't get a pairing request - baffled!

### DISCUSSION THREAD

**Discussion by CodyBradford** [NightscoutFoundation/xDrip#2472] [Q&A]
These discussions have helped me in the past. I've always managed to resolve any issues by reading through old posts, but this one has me stumped 🫤 

After working perfectly for months, I suddenly lost connection with my G6 transmitter. After double checking all my settings, I downloaded a BLE scanner and realized my transmitter wasn't transmitting. I installed a new sensor and transmitter, but couldn't initiate a pairing request. I tried changing the last digit of the transmitter serial then changing it back, but it wouldn't work. I could see it briefly connect every 5min, but then lose connection. I confirmed that the transmitter was working with the scanner, and that my phone was able to connect to other devices, but I couldn't get a connection through xdrip. I repeatedly got a message to enable Bluetooth, but it was always enabled. 

Navid was giving me suggestions of things to try, but nothing worked and eventually I had to connect to my old Dexcom receiver, which I'm still using - unfortunately. 

The phone is a Pixel 6 Pro, I've confirmed numerous times that my settings are identical to those recommended, and my Bluetooth and battery optimization settings are correct. I don't have any other Dexcom apps on my phone or any other devices that might be interfering. My setup was working perfectly for months, until suddenly it wasn't! 

![FB_IMG_1668640228137](https://user-images.githubusercontent.com/76657974/202314772-02a6ed5a-2562-4408-8976-86cdad66ebd1.jpg)

![1667601121917](https://user-images.githubusercontent.com/76657974/202314265-a03217c6-21f4-4fe0-9be7-6eddf59c4002.jpg)

**Comment by jamorham** [comment]
The only thing we really have to go on here is the reports that bluetooth is off in the log. Does this continue to appear? Are you able to test your phone's bluetooth and connect to anything else to confirm it is on and working? The only other thing that could block a pairing request is if do not disturb is enabled on the phone.

**Reply by CodyBradford** [reply]
As I mentioned in the post, my phone is able to connect to other devices just fine, even while the log was showing that Bluetooth was off. Bluetooth was never turned off, nor was Do Not Disturb enabled.

One other thing I just remembered: while I was trying to initiate a pairing request it would make the default noise that accompanies a pairing request, without a prompt showing up. I thought perhaps the Bluetooth notifications were disabled, but I checked and they aren't.

**Reply by Navid200** [reply]
> As I mentioned in the post, 

That's unnecessary to include in your response.  The person you are talking to has taken the time to look at this individual post and respond to it.  There is absolutely no way he can go through every single post and read every single word.  If he did that, xDrip would become a piece of garbage because he would have no time to do any technical work.  It's impossible for anyone to do all that.

It is understandable that you are frustrated.  But, you must understand that we are trying to help, and we are volunteers. 

<br/>  
<br/>  

> 
> One other thing I just remembered: while I was trying to initiate a pairing request it would make the default noise that accompanies a pairing request, without a prompt showing up. 

This is very important and you never mentioned it before.

Post a screenshot of Android notification settings for xDrip.

**Reply by CodyBradford** [reply]
> 
> That's unnecessary to include in your response. 
> 
Apologies, I wasn't trying to be combative.

> This is very important and you never mentioned it before.
> 

I only realized this recently, since our previous discussion on this issue 

All notifications are enabled

![Screenshot_20221117-143836](https://user-images.githubusercontent.com/76657974/202565321-003227e9-ee2c-4e78-9f7e-f96d5ac46231.png)

**Reply by Navid200** [reply]
Can you go to Android Settings.  Then, go to Sound.
Then, go to Do not Disturb.
Is there an option on the screen titled "Apps"?

**Reply by CodyBradford** [reply]
Yes, there is. I have xdrip set to interrupt DND

![Screenshot_20221117-144924](https://user-images.githubusercontent.com/76657974/202567179-519d95f6-3a78-4dad-bb3b-0007cb0bd19b.png)

**Reply by Navid200** [reply]
What do you see if you tap on this?
![202567179-519d95f6-3a78-4dad-bb3b-0007cb0bd19b](https://user-images.githubusercontent.com/51497406/202590341-3248fcaf-630e-4761-8f82-1eb209b4511f.png)

**Reply by rudysroost** [reply]
![Screenshot_20221117-202714](https://user-images.githubusercontent.com/12412982/202603007-bab625e4-77d3-4576-b05d-6c80b15fd6fb.png)

**Reply by CodyBradford** [reply]
Yes, this 👆
I took a screenshot but it wouldn't load

**Reply by Navid200** [reply]
I always tell everyone not to use Android Bluetooth settings page for pairing and wait for the notification.   But, this is a special case.

What happens if you bring up the Android Bluetooth page where you are searching for devices to link with.  Then, go back to xDrip and change the last transmitter ID character and back to trigger a pair request.
When you hear the sound, which you say does not bring up a notification, right away go to the Bluetooth page and search, if you see the transmitter, can you pair then?

**Reply by CodyBradford** [reply]
I'll give it a go. 

So my transmitter is now paired to my receiver. Can I just turn off the receiver and attempt to pair with xdrip? If the transmitter pairs to a different device will I need to restart the sensor?

**Reply by Navid200** [reply]
Looking at the system status page you have posted above, engineering mode is disabled.  That is great.  As long as you have not enabled engineering mode, xDrip will never interfere with the Dexcom receiver.  They use different transmitter slots.  So, you can leave the receiver active as is while you also establish connectivity from xDrip.

**Reply by Navid200** [reply]
Establishing connectivity from xDrip to a session in progress will never require doing anything to the sensor.  Please leave the sensor alone.

**Reply by Navid200** [reply]
https://navid200.github.io/xDrip/docs/Session-in-progress

**Reply by CodyBradford** [reply]
Got it, thanks. I'll try it

**Reply by CodyBradford** [reply]
Well dang, that worked. I didn't get a pairing request, just the blip noise, then I clicked it in the 'Pair New Device' list. It took a couple tries to get the timing right. 

Thanks Navid.

I wish I knew why I had to do it this way, but at least it's working, for now 🤨

![Screenshot_20221117-211451](https://user-images.githubusercontent.com/76657974/202615719-342b9d8c-3bb8-4743-9a0e-28cae4b39316.png)

**Reply by Navid200** [reply]
Have you set Bluetooth battery to unrestricted?

**Reply by Navid200** [reply]
Yes, this is unusual.  I would appreciate it if you don't suggest this to everyone who may have a connectivity issue.  Most people experiencing this problem will have it resolved if they follow the usual troubleshooting steps.

**Reply by CodyBradford** [reply]
Yes

![Screenshot_20221117-212349](https://user-images.githubusercontent.com/76657974/202616503-456aa3cb-cd66-4442-b5f9-cb6978025edc.png)

**Reply by Navid200** [reply]
OK,

It should be OK now.

**Reply by CodyBradford** [reply]
> Yes, this is unusual. I would appreciate it if you don't suggest this to everyone who may have a connectivity issue. Most people experiencing this problem will have it resolved if they follow the usual troubleshooting steps.

Over the years I've had numerous connectivity issues but this is the only one I haven't been able to resolve by reading through the documentation and double checking all the settings.

I think, like jamorham said, there's an issue with the Bluetooth itself. Since I upgraded to Android 13 I've had a few glitchy Bluetooth issues. I have a set of Bose headphones that won't connect, won't even show up in the devices list 🤷🏼‍♂️

**Reply by 1ethanhansen** [reply]
> I always tell everyone not to use Android Bluetooth settings page for pairing and wait for the notification. But, this is a special case.
> 
> What happens if you bring up the Android Bluetooth page where you are searching for devices to link with. Then, go back to xDrip and change the last transmitter ID character and back to trigger a pair request. When you hear the sound, which you say does not bring up a notification, right away go to the Bluetooth page and search, if you see the transmitter, can you pair then?

I'm on a Pixel 7 running Android 13, everything else failed troubleshooting-wise, but this worked for me, thank you!!

**Reply by rudysroost** [reply]
> I'm on a Pixel 7 running Android 13, everything else failed troubleshooting-wise, but this worked for me, thank you!!



@1ethanhansen Thank you for your message, it gave me hope as my device is the same as yours. Unfortunately, the solution does not work for me.

![unnamed](https://user-images.githubusercontent.com/12412982/204638853-13fd00db-edc6-44e3-9fc7-2baecd3df374.png)


![unnamed](https://user-images.githubusercontent.com/12412982/204638781-f4b54cff-c4c6-47d5-941f-fe460e1880a3.png)

**Reply by 1ethanhansen** [reply]
> @1ethanhansen Thank you for your message, it gave me hope as my device is the same as yours. Unfortunately, the solution does not work for me.
> 
> ![unnamed](https://user-images.githubusercontent.com/12412982/204638853-13fd00db-edc6-44e3-9fc7-2baecd3df374.png)
> 
> ![unnamed](https://user-images.githubusercontent.com/12412982/204638781-f4b54cff-c4c6-47d5-941f-fe460e1880a3.png)

Yeah that's about what I was experiencing too. Honestly not sure if this advice is what really did it because I was trying a lot of things too. Another thing I tried that might be the solution is I shut down my old phone (which was still getting data). The new phone started working about 10 minutes after that. Do you still have an old device getting data that you can shut down?

**Reply by rudysroost** [reply]
I do, and I did shut it down, but maybe not as long as you did. I'll try again after leaving the old phone off for 30 minutes or so.

**Comment by jamorham** [comment]
During a pairing request xDrip will make a blip noise in addition to any noise the pairing request would make via android system. This is specifically to attract attention. The only time I have seen the pairing dialog not appear when the sound plays is when do-not-disturb is enabled. Its possible there could be other situations, like if android system notifications were disabled or something else that would maybe cause the pairing request to appear in the notification shade but not actually pop up on screen.

**Reply by CodyBradford** [reply]
I've been trying to find a notification setting that might cause this, but I have all Bluetooth notifications turned on and I'm not sure what system notifications might interfere with this?

**Reply by jamorham** [reply]
I'm not sure what would cause it either. The pairing notifications don't come from xDrip though they would come from likely "Android System" - I didn't think the system notifications allowed themselves to be hidden but do not disturb definitely can hide them. If that wasn't enabled then I don't know why you wouldn't see them. Something looks sick with the bluetooth though based on the logs and this pairing problem. Is the pixel 6 running the latest standard android software? (Android 13 etc?)

**Reply by CodyBradford** [reply]
Agreed, perhaps this is beyond the scope of xdrip support! 

Yes, Android 13.

**Reply by rudysroost** [reply]
My very similar problems happened after my Pixel 5 got upgraded to Android 13. And then on my Pixel 7 also running Android 13. The Dexcom app can't connect either. I apologize if this post is inappropriate, I'm
 only trying to help the community, which includes me.

**Reply by CodyBradford** [reply]
> My very similar problems happened after my Pixel 5 got upgraded to Android 13. 

Interesting - mine seemed to work ok for a couple weeks after upgrading to 13, then this happened. Did you find a solution?

**Reply by rudysroost** [reply]
Mine did too, but kept connecting less frequently until it finally stopped completely. For a while, it would only connect once after rebooting the phone and never again until the next reboot. Then I switched to the Dexcom app. Same thing, consistent connections initially, then less and less frequent until it won't connect ever. Now, neither app will connect, not even once. 

My workaround is a cheap phone running Android 8 and following it with my Pixel.

**Reply by rudysroost** [reply]
@CodyBradford My Pixel 7 with Android 13 was still not connecting until yesterday. I got the December Android update, it took about five hours to install, but ever since the update, xDrip has been working perfectly! My phone has not missed a reading in over sixteen hours. 
Here is a list of all the issues fixed by the update: https://support.google.com/pixelphone/thread/191508547/google-pixel-update-december-2022?hl=en#:~:text=certain%20conditions%20*%5B2%5D-,Bluetooth,-Fix%20for%20issue
